src/dlgPackageManager.cpp

```diff
@@ -112,138 +235,381 @@ void dlgPackageManager::slot_installPackage()
+        //: Package manager - error when attempting to read a file to import
         QMessageBox::warning(this, tr("Import Mudlet Package:"), tr("Cannot read file %1:\n%2.").arg(fileName.toHtmlEscaped(), file.errorString()));
         return;
     }
 
     mpHost->installPackage(fileName, enums::PackageModuleType::Package);
 }
 
-void dlgPackageManager::slot_removePackages()
-{
-    const QModelIndexList selection = packageTable->selectionModel()->selectedRows();
-    QStringList removePackages;
-    for (const QModelIndex& index : selection) {
-        auto package = packageTable->item(index.row(), 0);
-        removePackages << package->text();
+void dlgPackageManager::slot_installPackageFromRepository()
+{ 
+    const QList<QListWidgetItem*> selected = packageList->selectedItems();
+    if (selected.isEmpty()) {
+        return;
     }
 
-    for (const QString& package : removePackages) {
-        mpHost->uninstallPackage(package, enums::PackageModuleType::Package);
+    this->setEnabled(false);
+
+    QNetworkAccessManager *manager = new QNetworkAccessManager(this);
+    QHash<QString, QString> *pendingDownloads = new QHash<QString, QString>();
+    int *remainingDownloads = new int(selected.size());
+    bool repoError = false;
+
+    for (QListWidgetItem* item : selected) {
+        const QString packageName = item->text();
+
+        QJsonObject foundObj;
+        for (const QJsonValue &val : repositoryPackages) {
+            QJsonObject obj = val.toObject();
+            if (obj.value("mpackage").toString() == packageName) {
+                foundObj = obj;
+                break;
+            }
+        }
+
+        if (foundObj.isEmpty()) {
+            repoError = true;
+            continue;
+        }
+
+        const QString remoteFileName = foundObj.value("filename").toString();
+        if (remoteFileName.isEmpty()) {
+            repoError = true;
+            continue;
+        }
+
+        const QByteArray encoded = QUrl::toPercentEncoding(remoteFileName);
+        const QUrl downloadUrl(qsl("https://github.com/Mudlet/mudlet-package-repository/raw/refs/heads/main/packages/%1").arg(QString::fromUtf8(encoded)));
+        const QString outDir = mudlet::getMudletPath(enums::profileHomePath, mpHost->getName());
+        const QString outPath = outDir + QDir::separator() + remoteFileName;
+
+        QNetworkReply *reply = manager->get(QNetworkRequest(downloadUrl));
+
+        QFile *file = new QFile(outPath);
+        if (!file->open(QIODevice::WriteOnly)) {
+            file->deleteLater();
+            reply->deleteLater();
+            continue;
+        }
+
+        QObject::connect(reply, &QNetworkReply::readyRead, [file, reply]() {
+            file->write(reply->readAll());
+        });
+
+        pendingDownloads->insert(packageName, outPath);
+
+        QObject::connect(reply, &QNetworkReply::finished, [reply, file, this, outPath, packageName, pendingDownloads, remainingDownloads, manager]() {
```

Comment:
Locations: Lines 123, 145, 303

  Problem: Lambdas capture this for callbacks, but if the dialog closes during network operations, this becomes a dangling pointer. When Qt invokes the callback later, it will access deleted memory, causing crashes.

  Example from line 303:
  QObject::connect(reply, &QNetworkReply::finished, [reply, file, this, outPath, ...] {
      // If dialog closed, 'this' is dangling
      this->setEnabled(true); // CRASH

  Fix: Use `QPointer<dlgPackageManager>` (preferred) or connect with context object: `connect(reply, &QNetworkReply::finished, this, [...]() {...});`